pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -883,6 +897,35 @@ func endpointDiscoveryResponse(loadAssignments []*xdsapi.ClusterLoadAssignment,
 	return out
 }
 
+func endpointAcceptMtls(endpointTlsMap *map[string]bool, push *model.PushContext, ep *model.IstioEndpoint) bool {
+	b, err := json.Marshal(ep.Labels)
+	if err != nil {
+		return true
+	}
+	labelHash := fmt.Sprintf("%x", sha256.Sum256(b))[:8]
+	key := fmt.Sprintf("%v|%v|%v", ep.Attributes.Namespace, labelHash, ep.EndpointPort)
```

Comment:
Why not use the uid here

Reply:
uid seems to be one to one mapping to a pod id? `https://github.com/istio/istio/blob/974a8770fdbe2a2a55599b0a9296f2934ab1a852/pilot/pkg/serviceregistry/kube/controller/endpoints.go#L175`

this way, two pods from the same deployment, even though they have same label set, we will generate different uid, thus cache can't be hit.

Reply:
The labels is not same for them

Reply:
hmm, they are the same right? i mean if i create `httpbin` deployment, specifying `replica: 2`, both pods should have the same labels. no?